### PR TITLE
Register external asset addresses to account state

### PR DESCRIPTION
**Pull request by lmahanand**
## Problem
Currently we are only registering the address of HER token and we are not registering the external asset addresses.
Asana Link: 
[Problem description](https://app.asana.com/0/998359196895599/1123163206272529)
## Testing Done and Results
1. Tested registration of HER Address.
2. Tested registration of ETH Address
3. Executed make run-test with all test cases passing
4. Executed the command `curl http://localhost/account/HHy1CuT3UxCGJ3BHydLEvR5ut6HRy2qUvm` and below is its result

Screenshot 2019-05-15 at 8 02 01 PM

### CODE REVIEW

File: hbi/message/plugin.go [hunk 1]
```diff
@@ -371,15 +372,29 @@ func getAccount(address string, ctx *network.PluginContext) error {
 	}
 
 	if account != nil {
+		eBalances := make(map[string]*protoplugin.EBalance)
+
+		if account.EBalances != nil && len(account.EBalances) > 0 {
+			for key := range account.EBalances {
```
Comment (bittelc) [inline hbi/message/plugin.go]: Do we need any `key` validation here? As in, do we want to only allow `ETH` and `BTC`?

File: hbi/protobuf/service.proto [hunk 2]
```diff
@@ -46,6 +48,10 @@ message Asset {
   uint64 value            = 4;
   uint64 fee              = 5;
   uint64 nonce            = 6;
+  string external_sender_address    = 7;
```
Comment (bittelc) [inline hbi/protobuf/service.proto]: Why are these needed? And why don't they go into the `eBalance` struct. As they are now, they're just clutter when the transaction is a local `HER` => `HER` tx.

**Comment by lmahanand** [comment]
> We need tests!

Oh yes.. I will put some test cases.

File: hbi/protobuf/service.proto (hunk 2 above)
Comment (lmahanand) [reply hbi/protobuf/service.proto]: We can't put them in eBalance struct since it is required only to hold the account states of external assets. However, we have external_reciever_address along with external_sender_address that will be required to update the state of receiver's account holding external assets.

Below fields are required to be sent as a part of TX payload.
``
string external_sender_address    = 7;
string external_reciever_address  = 8;
uint64 external_nonce             = 9;
uint64 external_block_height      = 10;
``

File: hbi/message/plugin.go (hunk 1 above)
Comment (lmahanand) [reply hbi/message/plugin.go]: No. It is not required since the values will be updated by account update process being built and integrated by @niktrix . And we can directly retrieve the values against the keys and populate them in account response to be sent back to API node.
